### DISCUSSION TITLE
Feature suggestion: download only a SEGMENT of an online video

### DISCUSSION THREAD

**Discussion by stevenkain** [aclap-dev/video-downloadhelper#1136] [Help and Q&A]
### Your question

Usually I find segments of a clip or quote just a few seconds long in an online video of 4-7 hours long. Don't want to download it all, just that specific, short segment of it.
Right now I download it (with downloadhelper) then use a dedicated app to roughly cut it to size, then upload it to an online video editor to cut it again, more accurately, and convert it to a chat-app compatible format. Would be great to be able to do all this in just 1 click, even pay for such a feature.

### Addon & CoApp version and details

feature request, not complaint.

**Comment by paulrouget** [comment]
Not the first time it's suggested. Also make sense for live streams. It's been in my todo list for a while now 😅 need to find the time.

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

ffmpeg -ss hh:mm:ss -t hh:mm:ss -i inputsource targetfile

-ss skips the specified duration at the start of the inputsource, which might be a livestream, a saved video, even a file on your system.

-t limits the duration of the inputsource that gets recorded to targetfile.

You can probably also insert this immediately before targetfile:

-codec: copy

You will probably want to experiment with all of this.  I'm not entirely certain how -ss will be accepted by web sites.  I believe -- I haven't actually done this with content from the web so this is just a belief -- that the recognition of -ss depends on the way the web server is implemented.  I believe there is some configuring the webmaster has to do for -ss to actually skip some offset into an inputsource.  I also don't know if -ss is efficient or a terrible waste of resources.  I can imagine that in order to process -ss correctly, the server might have to read the entirety of the part you're trying to skip in order to arrive at the part you actually want to read.  Or the server might be smart enough to skip directly to your desired starting point.  You'll have to experiment with this.  What I'm trying to say is -ss might be accepted by the web server & it might simply be ignored.  You just have to find out on a case by case basis.  At the worst, you can download whatever it is you are downloading, then process your downloaded file on your system as the input object I am calling inputsource.

If this is all beyond your understanding, then don't bother.  But if you are willing to experiment with this, do please come back to tell us your experience.

**Reply by Mattmjs** [reply]
I've done that and it works. Another option is using VLC media player and using the m3u8 url. Before the download is started
you can specify a start and finish time.
